### PR TITLE
BTFS-937-release add docker-compose to build binaries

### PR DESCRIPTION
**Pull request by richlee33**
here is the PR to merge to release branch.  jenkins will run docker-compose to build binaries to eliminate variability in the version of go.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: '3'
+services:
+  btfs-build-binaries:
+    build:
+      context: .
+      dockerfile: Dockerfile.unit_testing
+    volumes:
+      - ../btfs-binary-releases:/btfs-binary-releases
```
Comment (taiyangc) [inline docker-compose.yml]: This looks like a path on jenkins, have you tested docker-compose manually on there?

Comment (richlee33) [reply docker-compose.yml]: i tested it by running the jenkins job on my dev branch and it looked fine:
https://builds.bittorrent.com/job/BTFS_Build_Binary/106/

**Comment by taiyangc** [comment]
Need another approval since this is going to `release`

File: docker-compose.yml (hunk 1 above)
Comment (wxue) [inline docker-compose.yml]: Can you help to fill me some background info here -- why copy previous container's volume to the new?

Comment (richlee33) [reply docker-compose.yml]: when running the docker container to build the binaries, we need to share a directory on the jenkins slave so the binaries built by the container will persist and can be uploaded to the s3 bucket as the last step of the jenkins job.
when docker-compose runs .`/package.sh` in the go-btfs dir, this script places the built binaries in the btfs-binary-releases directory that is at the same level as the go-btfs dir.
we dont need to necessarily copy the files from ../btfs-binary-releases into the container but it needs to be accessable by the container as a place to save the newly generated binaries.

Comment (wxue) [reply docker-compose.yml]: I see, thanks for the info.

**Pull request merged**